app/validators/program_validator.rb

```diff
@@ -1,17 +1,36 @@
 # frozen_string_literal: true
 
 class ProgramValidator < BaseValidator
-  def validate(record)
-    duplicates_exist = Program.where(['description = ? AND facility_code = ? AND id != ?',
-                                      record.description, record.facility_code, record.id]).any?
-    record.errors[:base] << non_unique_error_msg(record) if duplicates_exist
+  def self.after_import_batch_validations(validation_warnings)
```

Comment:
Since we changed back to AR objects, can we rename this parameter `failed_instances`?

Reply:
changed